rewrite commercial support policy

juliandunn opened the pull request:
This PR rewrites the commercial support policy for Chef's software. There are several objectives in doing so:

* Explicitly delineate commercial and community support (in essentially all situations, commercial support of platforms is broader and better than community support)
* Remove the notion of "foundational" and "secondary" which are confusing to customers as they do not care what testing rigor we apply to things we support, only that there is a mean-time-to-respond SLA associated with a commercial support contract.

It's important to call out several changes in this PR. As written, it commits Chef Software Inc. to commercial support of several additional platforms and versions. However, it is my belief that these reflect market realities and so we must be responsive to that.

* Chef Client: I largely merged "foundational" and "secondary" and also:
  * Added AIX 7.2 to regular support
  * Added RHEL and SLES `s390x`, `ppc64le` and `ppc64be` to mainstream support
  * Added Amazon Linux as mainstream support (for possible discussion)
  * Added Windows 2016 as mainstream support
  * Moved Debian 7.x and 8.x to community support
  * Removed mention of all extraneous platforms that we don't support (NetBSD, HP-UX, etc.) and that the community doesn't support either. In other words: if it's not explicitly mentioned in this doc, an OS doesn't even get community support.

* Chef Automate
  * Removed minor versions from EL6 platforms
  * Added Oracle Enterprise Linux as commercially-supported

* Chef Automate Runners
  * Cleaned up prose and put it in a different section

* Chef Push Jobs Client
  * Edited to make it clear we're talking about the push jobs client

Please also review the table at the bottom that describes our support policy, how we align with vendors' OS EoLs, and explicitly indicates when we stop building software for certain variants.

Commit:
Draft revised commercial support policy for discussion.

Signed-off-by: Julian C. Dunn <jdunn@chef.io>

brewn commented:
Leaving this here so I remember it - whenever this guide gets updated, we should remove the platforms stuff from this guide and incorporate it here: https://docs.chef.io/install_chef_compliance.html

Commit:
Removed "mainstream" versus "extended" -- we are not publicizing an extended support
offering at this time.
Include tables that illustrate our EoL policy as aligned with vendors, as well as to
callout exactly what OSes go EoL when.

Signed-off-by: Julian C. Dunn <jdunn@chef.io>

juliandunn commented:
@scotthain this is now ready for your review.

Commit:
replicate tags for consistency

Signed-off-by: Julian C. Dunn <jdunn@chef.io>

scotthain commented on chef_master/source/platforms.rst:
```diff
@@ -35,38 +33,41 @@ The following table lists the Foundational platforms for the chef-client and oha
+     - ``i386`` (5.x only), ``x86_64``
      - ``5``, ``6``, ``7``
    * - FreeBSD
      - ``i386``, ``amd64``
```
As an aside, we only test on 10-i386, not 11-i386

juliandunn replied:
I assume we want to write it as FreeBSD `i386` is supported only on `10-STABLE` and for `11-STABLE` we would want to only support `amd64`?

scotthain commented on chef_master/source/platforms.rst:
```diff
@@ -35,38 +33,41 @@ The following table lists the Foundational platforms for the chef-client and oha
-     - ``i386``, ``x86_64``
-     - ``5``, ``6``, ``7``
+     - ``i386`` (5.x only), ``x86_64``, ``s390x``, ``ppc64le``, ``ppc64be``
+     - ``6``, ``7``
```
Previous line references 5.x - also ppc64 is only 7

scotthain commented on chef_master/source/platforms.rst:
```diff
@@ -35,38 +33,41 @@ The following table lists the Foundational platforms for the chef-client and oha
+     - ``x86_64``, ``s390x``, ``ppc64le``, ``ppc64be``
      - ``11 SP4``, ``12 SP1``
-   * - Ubuntu
+   * - Ubuntu (LTS releases)
```
Also 14.04 ppc64le (not be)

scotthain commented on chef_master/source/platforms.rst:
```diff
@@ -172,9 +116,6 @@ The following table lists the Foundational platforms for the Chef DK:
      - Version
-   * - Debian
-     -
-     - ``7.x``, ``8.x``
```
Does this mean we can stop building chefdk on debian? :D

juliandunn replied:
Maybe? Do we know how this even started? Like what customer needed it? (you can message me privately if you have that info, since this is a public PR)

scotthain commented on chef_master/source/platforms.rst:
```diff
@@ -332,9 +256,6 @@ The following table lists the Foundational platforms for the Chef push jobs:
-     - ``i386``, ``x86_64``
-     - ``7``
    * - Red Hat Enterprise Linux
      - ``i386``
```
push jobs client also builds/tests on x86_64 for all platforms, fyi. (and OSX x86_64)

scotthain commented on chef_master/source/push_jobs.rst:
```diff
@@ -44,9 +44,6 @@ The following table lists the Foundational platforms for the Chef push jobs:
-     - ``i386``, ``x86_64``
-     - ``7``
    * - Red Hat Enterprise Linux
      - ``i386``
```
Same as above comment.

Commit:
Address review comments. Also:

* Remove 32-bit FreeBSD from support
* Can't remove EL5.x from support yet until end-of-year, so be consistent
* Remove all Ubuntu 12.04 as we are EOLing support for that immediately
* Added a few missing products that we didn't have documentation for

Signed-off-by: Julian C. Dunn <jdunn@chef.io>

juliandunn committed:
Merge branch 'master' into rewrite-support-policy

juliandunn committed:
Merge branch 'master' into rewrite-support-policy

juliandunn commented:
Ok @scotthain could use a re-review based on our discussion. To wit:

* We remove 32-bit FreeBSD from support at all
* Need to leave all mentions of EL5 in-place until the end-of-year to be consistent with the table
* Remove Ubuntu 12.04 as we're EOLing that immediately

I also added tables for products that were missing.

Also fixed up a bunch of stylistic nits (like we should say 5.x, 6.x, 7.x everywhere for EL platforms)

scotthain commented on chef_master/source/platforms.rst:
```diff
@@ -176,13 +177,13 @@ The following table lists the commercially-supported platforms and versions for
      - ``5.x``, ``6.x``, ``7.x``
    * - Red Hat Enterprise Linux
-     - ``x86_64``
+     - ``x86_64``, ``ppc64`` (7.x only), ``ppc64`` (7.x only)
```
missing an `le` on there.

scotthain reviewed (changes requested):
One more tiny nitpick

Commit:
Missed "le"

Signed-off-by: Julian C. Dunn <jdunn@chef.io>

Commit:
Merge branch 'rewrite-support-policy' of github.com:juliandunn/chef-web-docs into rewrite-support-policy

juliandunn commented:
@brewn this is ready to go whenever you are ready.
